src/resnext.jl

```diff
@@ -0,0 +1,117 @@
+"""
+    resnextblock(inplanes, outplanes, cardinality, downsample = false)
+
+Create a basic residual block as defined in the paper for ResNeXt
+([reference](https://arxiv.org/abs/1611.05431)).
+
+# Arguments:
+- `inplanes`: the number of input feature maps
+- `outplanes`: the number of output feature maps 
+- `cardinality`: the number of groups to use for the convolution
+- `width`: the number of feature maps in each group in the bottleneck
+- `downsample`: set to `true` to downsample the input
+"""
+function resnextblock(inplanes, outplanes, cardinality, width, widen_factor, downsample = false)
+    stride = downsample ? 2 : 1
+
+    width_ratio = outplanes / (widen_factor * 128.0)
+    hidden_channels = cardinality * floor(width * width_ratio)
+
+    return Chain(conv_bn((1, 1), inplanes, hidden_channels; stride = 1, bias = false)...,
+        conv_bn((3, 3), hidden_channels, hidden_channels;
+            stride = stride, pad = 1, bias = false, groups = cardinality)...,
+        conv_bn((1, 1), hidden_channels, outplanes; stride = 1, bias = false)...)
+end
```

Comment:
I don't believe the `widen_factor` should be considered here. The inner channel dimension is always `cardinality * width`, and it is independent of the input and output channels. See Fig. 3c in the paper: the input and output channels are 256, and the inner channels is 32*4 = 128.

Reply:
That doesn't really match the image though. In that case, the inner channels should always be 128, whereas the image explicitly shows the inner channels increasing for every block. And plus that configuration does not match the parameters specified in the paper, only this one does. The params and FLOPs line up with this specific implementation. But let me know if I'm missing something

Reply:
Oh you're right, I was completely focused on the figures, but the table does indeed have an increasing inner dimension. What I would suggest then is still keeping the `inner = cardinality * width` approach since that matches how the bottleneck is described in the paper. Then in main `resnext` function, we can widen the `width` passed in by `widen_factor` just like we do for `baseplanes`.